app/components/galoy-text-input/galoy-text-input.tsx

```diff
@@ -0,0 +1,110 @@
+import { palette } from "@app/theme"
+import React, { useEffect, useState } from "react"
+import { TextInput, View } from "react-native"
+import { Text } from "react-native-elements"
+import EStyleSheet from "react-native-extended-stylesheet"
+
+type GaloyTextInputProps = {
+  placeholder?: string
+  append?: string
+  prepend?: string
+  validation?: [() => boolean]
+  label: string
+  onChangeText: (text: string) => void
+  labelStyle?: EStyleSheet.AnyObject
+}
+
+const styles = EStyleSheet.create({
+  container: {
+    flexDirection: "column",
+  },
+  labelContainer: {
+    flex: 1,
+  },
+  inputContainer: {
+    flex: 1,
+  },
+  label: {
+    color: palette.inputLabel,
+    fontSize: 18,
+    lineHeight: 24,
+    fontFamily: "Roboto",
+    fontWeight: "500",
+  },
+  input: {
+    backgroundColor: palette.inputBackground,
+    borderRadius: 8,
+    paddingVertical: 8,
+    paddingHorizontal: 12,
+  },
+  appendTextStyle: {
+    color: palette.secondaryText,
+  },
+  prependTextStyle: {
+    color: palette.secondaryText,
+  },
+})
+
+export const GaloyTextInput = ({
+  placeholder,
+  label,
+  labelStyle,
+  append = "",
+  prepend = "",
+  onChangeText,
+}: GaloyTextInputProps) => {
+  const [value, setValue] = React.useState("")
+  const [cleanValue, setCleanValue] = useState("")
```

Comment:
I think this may be an unnecessary duplication of state, because the two values are deterministically related. The ```value``` can be functionally derived by simply adding the prepend and/or append to the ```cleanValue```.  The risk would be that any time state is duplicated it could introduce bugs and complexity surrounding keeping them in sync.